src/reactviews/pages/TableDesigner/designerChangesPreviewButton.tsx

```diff
@@ -130,13 +146,48 @@ export const DesignerChangesPreviewButton = () => {
                 </>
             );
         }
+        if (metadata?.apiState?.publishState === LoadState.Error) {
```

Comment:
these LoadState cases feel like they're screaming for refactoring the dialog content out into helper functions/components.